test/mocking_utils/patch.hpp

```diff
@@ -0,0 +1,218 @@
+  typename ArgType4, typename ArgType5>
+struct PatchTraits<N, ReturnType(ArgType0, ArgType1, ArgType2, ArgType3, ArgType4, ArgType5)>
+{
+  mmk_mock_define(
+    mock_type, ReturnType, ArgType0, ArgType1, ArgType2, ArgType3, ArgType4, ArgType5);
+};
+
+template<size_t N, typename SignatureT>
+struct Trampoline;
+
+template<size_t N, typename ReturnT, typename ... ArgTs>
+struct Trampoline<N, ReturnT(ArgTs...)>
+{
+  static ReturnT base(ArgTs... args)
+  {
+    return target(std::forward<ArgTs>(args)...);
+  }
+
+  static std::function<ReturnT(ArgTs...)> target;
+};
+
+template<size_t N, typename ReturnT, typename ... ArgTs>
+std::function<ReturnT(ArgTs...)>
+Trampoline<N, ReturnT(ArgTs...)>::target;
+
+template<size_t N, typename SignatureT>
+auto prepare_trampoline(std::function<SignatureT> target)
+{
+  Trampoline<N, SignatureT>::target = target;
+  return Trampoline<N, SignatureT>::base;
+}
+
+template<size_t N, typename SignatureT>
+class Patch;
+
+template<size_t N, typename ReturnT, typename ... ArgTs>
+class Patch<N, ReturnT(ArgTs...)>
+{
+public:
+  using mock_type = typename PatchTraits<N, ReturnT(ArgTs...)>::mock_type;
+
+  Patch(const std::string & target, std::function<ReturnT(ArgTs...)> proxy)
+  : proxy_(proxy)
+  {
+    auto MMK_MANGLE(mock_type, create) =
+      PatchTraits<N, ReturnT(ArgTs...)>::MMK_MANGLE(mock_type, create);
+    mock_ = mmk_mock(target.c_str(), mock_type);
+  }
+
+  Patch(const Patch &) = delete;
+  Patch & operator=(const Patch &) = delete;
+
+  Patch(Patch && other)
+  {
+    mock_ = other.mock_;
+    other.mock_ = nullptr;
+  }
+
+  Patch & operator=(Patch && other)
+  {
+    if (mock_) {
+      mmk_reset(mock_);
+    }
+    mock_ = other.mock_;
+    other.mock_ = nullptr;
+  }
+
+  ~Patch()
+  {
+    if (mock_) {
+      mmk_reset(mock_);
+    }
+  }
+
+  Patch & then_call(std::function<ReturnT(ArgTs...)> replacement) &
+  {
+    auto type_erased_trampoline =
+      reinterpret_cast<mmk_fn>(prepare_trampoline<N>(replacement));
+    mmk_when(proxy_(any<ArgTs>()...), .then_call = type_erased_trampoline);
+    return *this;
+  }
+
+  Patch && then_call(std::function<ReturnT(ArgTs...)> replacement) &&
+  {
+    auto type_erased_trampoline =
+      reinterpret_cast<mmk_fn>(prepare_trampoline<N>(replacement));
+    mmk_when(proxy_(any<ArgTs>()...), .then_call = type_erased_trampoline);
+    return std::move(*this);
+  }
+
+private:
+  template<typename T>
+  T any() {return mmk_any(T);}
+
+  mock_type mock_;
+  std::function<ReturnT(ArgTs...)> proxy_;
+};
+
+template<size_t N, typename SignatureT>
+auto make_patch(const std::string & target, std::function<SignatureT> proxy)
+{
+  return Patch<N, SignatureT>(target, proxy);
+}
+
+/// Define a dummy operator `op` for a given `type`.
+///
+/// Useful to enable patching functions that take arguments whose types
+/// do not define basic comparison operators required by Mimick.
+#define MOCKING_UTILS_DEFINE_DUMMY_OPERATOR(type, op) \
```

Comment:
suggest: MOCKING_UTILS_BOOL_OPERATOR_RETURNS_FALSE

Just so when people see it in code, it's a bit clearer what this macro is doing.

Reply:
See https://github.com/ros2/rcutils/pull/272/commits/83140ecc180b40aa9aff5e997d33cb19679d1b10.